### PR TITLE
Friendships v1

### PR DESCRIPTION
**Pull request by carlos-ssh**
**Hi there, in this branch i worked on the following features:**
- Friend invitation.
- Cancel a friend invitation.
- Decline friend invitation.
- Accept a friend's invitation.
- Delete friend
- Display all friend's posts
- Friendship model, friendships controller, and frienship_helper were created to achieve all these features.

Greetings.

Carlos-SSH

### CODE REVIEW

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @carlos-ssh 
Such a remarkable project you made :+1: ! Awesome README :100: !
We, however, do have a few changes to carry out before we proceed (see review comment).

##

Kindly work on the suggested changes  👇.
- [ ] `rails db:migrate` is not working, please check your foreign keys and columns
- [ ] View Demo in the README file redirecting to the repo, please remove it or update the live link

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by carlos-ssh** [comment]
Fixed the next requested issues:
- rails db:migrate, ( foreign keys and columns)
- View Demo Locked in the README file.

Greetings 
Carlos SSH

**Review by layzrd — CHANGES_REQUESTED** [review]
### Status: Require changes 🌳 ♲
Hi @carlos-ssh, I am Abdulaziz Redi Ali, your TSE for this PR.  👏👏  I appreciate your hard work.   Kindly fix the following issues.

[REQUIRED]

- Add unit and integration tests
- Remove unused commented gems from the Gemfile
- Activate linters workflow within your project

Once you fix the issues, kindly submit another code review. ♲

Happy coding 👨‍💻
[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

**Comment by carlos-ssh** [comment]
Fixed requested issues:
- Add unit and integration tests
- Remove unused commented gems from the Gemfile
- Activate linters workflow within your project

**Comment by mimipeshy** [comment]
### STATUS : CHANGES REQUESTED :red_circle: :recycle: 

Hey Carlos :smile: :wave: great to see you buddy :wave: Good job on working on the feedback given by the TSE's. However, you have not done the following

- [ ] You have not written actual tests for your code according to this ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) , 
KIndly write them and submit them for review.

That's all :smile: kindly submit another review when done.

_Feel free to reach out to me on slack if you have any questions or something is unclear, I will be glad to help_
Regards
Peris Ndanu

**Comment by carlos-ssh** [comment]
Worked on requested changes:
- Create tests with Rspec

File: Rakefile [hunk 1]
```diff
@@ -2,5 +2,6 @@
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
 require_relative 'config/application'
+require 'single_test/tasks'
```
Comment (SpaYco) [inline Rakefile]: - [x] `require 'single_test/tasks'` is giving me an error and not letting me migrate, I removed it and was able to migrate, maybe you have missing gems or this line isn't doing anything? please check it.

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] check below 👇 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

File: Rakefile (hunk 1 above)
Comment (carlos-ssh) [reply Rakefile]: it's done.

**Comment by carlos-ssh** [comment]
fixed issues with single taks requested.

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**